### DISCUSSION TITLE
[feature-request] Adapt S3 method `conditional_effects` to `bmmfit` objects.

### DISCUSSION THREAD

**Discussion by GidonFrischkorn** [venpopov/bmm#200] [Ideas]
The `conditional_effects` methods works ok for most `bmmodels`, but I think we could add some tweaks to integrate it more seamlessly into the workflow of `bmm`. For example, currently users need to know explicitly if a model parameter is specified as a `dpar` or `nlpar` when calling `conditional_effects`. Moreover, for `dpar` the `conditional_effects` are transformed back to the native scale using the specified links, this is not the case for `nlpars`.

Here are some examples, for `conditional_effects` plots for the two-parameter mixture model.

`conditional_effects(ZL_fit, effects = "setsize", dpar = "kappa1")` yields
![image](https://github.com/venpopov/bmm/assets/17825471/89e55871-5156-4858-a50d-4a547c90b421)

whereas `conditional_effects(ZL_fit, effects = "setsize", nlpar = "kappa")` yields:
![image](https://github.com/venpopov/bmm/assets/17825471/dc36a762-4d23-459c-884c-7861669d3d76)

For `theta1` versus `thetat` i thought that having the plot on the probability scale is super useful, but again that only works when calling it via the `dpar`:  `conditional_effects(ZL_fit, effects = "setsize", dpar = "theta1")`
![image](https://github.com/venpopov/bmm/assets/17825471/334860ef-72de-4daa-926a-d0d89f6763d4)

when called using the `nlpar`:  `conditional_effects(ZL_fit, effects = "setsize", nlpar = "thetat")`
![image](https://github.com/venpopov/bmm/assets/17825471/4b74f1cb-c9a1-454b-8fb0-28a56272acd0)

Ideally, we have an implementation of `conditional_effects` for `bmmfit` objects that uses a `par` argument to specify for which model parameter the effects should be plotted and maybe add an additional `scale` argument to specify if the plots should be given on the `parameter`/`sampling` scale or the `native` scale. 

What do you think @venpopov?

**Comment by venpopov** [comment]
This would be great to have. Do you want to take a stab at it?

**Reply by GidonFrischkorn** [reply]
Yes, I can take a look at this. I hope that I will find time next week. Probably, we can recycle a lot of functionality from the `conditional_effects.brmsfit` method.

**Comment by GidonFrischkorn** [comment]
Closed by #328
